Cython/Debugger/Cygdb.py

```diff
@@ -104,67 +104,62 @@ def make_command_file(path_to_debug_info, prefix_code='',
 
     return tempfilename
 
-usage = "Usage: cygdb [options] [PATH [-- GDB_ARGUMENTS]]"
 
-def main(path_to_debug_info=None, gdb_argv=None, no_import=False):
+def main():
     """
     Start the Cython debugger. This tells gdb to import the Cython and Python
     extensions (libcython.py and libpython.py) and it enables gdb's pending
     breakpoints.
-
-    path_to_debug_info is the path to the Cython build directory
-    gdb_argv is the list of options to gdb
-    no_import tells cygdb whether it should import debug information
     """
-    parser = optparse.OptionParser(usage=usage)
-    parser.add_option("--gdb-executable",
+    parser = argparse.ArgumentParser(
+        prog="cygdb",
+        description="Cython debugger",
+    )
+    parser.add_argument("gdb_argv", nargs="*",
+                        help="arguments to forward to gdb; specified after --")
+    parser.add_argument("--build_dir", default=os.curdir,
+                        help="directory containing cython_build/ files")
+    parser.add_argument("--no_import", default=False,
+                        help="whether cygdb should import debug information")
+    parser.add_argument("--gdb-executable",
         dest="gdb", default='gdb',
         help="gdb executable to use [default: gdb]")
-    parser.add_option("--verbose", "-v",
+    parser.add_argument("--verbose", "-v",
         dest="verbosity", action="count", default=0,
         help="Verbose mode. Multiple -v options increase the verbosity")
-    parser.add_option("--skip-interpreter",
+    parser.add_argument("--skip-interpreter",
                       dest="skip_interpreter", default=False, action="store_true",
                       help="Do not automatically point GDB to the same interpreter "
                            "used to generate debugging information")
 
-    (options, args) = parser.parse_args()
-    if path_to_debug_info is None:
-        if len(args) > 1:
-            path_to_debug_info = args[0]
-        else:
-            path_to_debug_info = os.curdir
-
-    if gdb_argv is None:
-        gdb_argv = args[1:]
+    parsed = parser.parse_args()
```

Comment:
TBH — and this is _heavy_ bikeshedding, admittedly — I really dislike `parsed` as a name for the results of `ArgumentParser.parse_args()`.

`parsed` represents the **state** of the results, not the **contents**, which is focusing on the wrong thing. The fact that the contents were _parsed_ is irrelevant compared to the question of **what the contents _are_**. But the name doesn't even _hint_ at that.

`options` is still pretty common when using `argparse`, same as it was with `optparse`. But if you don't like `options`... well, if you call a parser's `parse_args()` method, it seems pretty tautological what that would produce: `args`.

(You even label them exactly that way on line 152.):

```python3
logger.debug("args = %r", parsed)
```

So even your own code is subconsciously agreeing that `parsed` should be named `args`. :wink:

Reply:
Sure no strong attachment to `parsed`. The downside to `args` is that it conflicts with the pdb args command, so you have to jump through more hoops to inspect the object when debugging. Happy to revert back to options

Reply:
:+1: Thanks, `options` makes sense to me.

It struck me after posting my original comment: _another_ reason to avoid `parsed` is that having `parsed` and `parser` in the same scope (differing by only the last letter) feels like a recipe for long-term maintainability issues. Someone down the road could _easily_ confuse the two.